components/raftstore-v2/src/operation/misc.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::{
+    Bound::{Excluded, Unbounded},
+    HashSet,
+};
+
+use engine_traits::{KvEngine, RaftEngine, CF_DEFAULT, CF_WRITE};
+use slog::{debug, error, info};
+
+use crate::{
+    fsm::StoreFsmDelegate, router::StoreTick, worker::cleanup, CompactTask::CheckAndCompact,
+};
+
+impl<'a, EK: KvEngine, ER: RaftEngine, T> StoreFsmDelegate<'a, EK, ER, T> {
+    pub fn register_compact_check_tick(&mut self) {
+        self.schedule_tick(
+            StoreTick::CompactCheck,
+            self.store_ctx.cfg.region_compact_check_interval.0,
+        )
+    }
+
+    pub fn on_compact_check_tick(&mut self) {
+        self.register_compact_check_tick();
+        if self.store_ctx.schedulers.cleanup.is_busy() {
+            info!(
+                self.store_ctx.logger,
+                "compact worker is busy, check space redundancy next time";
+            );
+            return;
+        }
+
+        // Start from last checked key.
+        let mut regions_to_check: HashSet<u64> = HashSet::default();
+
+        let (largest_key, last_check_key) = {
+            let mut last_check_key = self.fsm.store.last_compact_checked_key();
+
+            let meta = self.store_ctx.store_meta.lock().unwrap();
+            if meta.region_ranges.is_empty() {
+                debug!(
+                    self.store_ctx.logger,
+                    "there is no range need to check";
+                );
+                return;
+            }
+            // Collect continuous ranges.
+            let ranges = meta.region_ranges.range((
+                Excluded((last_check_key.clone(), u64::MAX)),
+                Unbounded::<(Vec<u8>, u64)>,
+            ));
+
+            for region_range in ranges {
+                last_check_key = &region_range.0.0;
+                regions_to_check.insert(*region_range.1);
+
+                if regions_to_check.len()
+                    >= self.store_ctx.cfg.region_compact_check_step_v2 as usize
```

Comment:
what about reuse the v1's config? They serve the same purpose and won't be used in parallel.

Reply:
Resuing and adding some description about it would be fine.